beetsplug/info.py

```diff
@@ -98,8 +98,9 @@ def print_data(data, item=None, fmt=None):
     formatted = {}
     for key, value in data.items():
         if isinstance(value, list):
-            formatted[key] = u'; '.join(value)
-        if value is not None:
+            if value:
+                formatted[key] = u'; '.join(value)
+        elif value is not None:
```

Comment:
I understand what you mean. Note that list fields (genre, arranger ... ) now are always lists, the problem comes with code like that which tries to work with all fields at once. Would you prefer that I create a class that inherit from lists and reimplements __str__ so it returns '; '.join(self) ? That way code like that could be changed to:
```
if value:
    formatted[key] = str(value)
```
But still, that's not perfect neither.